sigar_interface_test.go

```diff
@@ -211,13 +212,51 @@ var _ = Describe("Sigar", func() {
 	})
 
 	It("proc time", func() {
-		time := ProcTime{}
-		err := time.Get(os.Getppid())
+		// Measure parent process
+		timeParent := ProcTime{}
+		err := timeParent.Get(os.Getppid())
 		Expect(err).ToNot(HaveOccurred())
-		Expect(time.User).To(BeNumerically(">", 0))
-		Expect(time.Sys).To(BeNumerically(">", 0))
+		Expect(timeParent.User).To(BeNumerically(">", 0))
+		Expect(timeParent.Sys).To(BeNumerically(">", 0))
 
-		err = time.Get(invalidPid)
+		// Take snapshot of current process
+		time1 := ProcTime{}
+		err = time1.Get(os.Getpid())
+		Expect(err).ToNot(HaveOccurred())
+
+		// Burn one second of CPU time
+		timer := time.NewTimer(time.Second)
+		var i int
+	forLoop:
+		for {
+			select {
+			case <-timer.C:
+				break forLoop
+			default:
+				i++
+			}
+		}
+		Expect(i).To(BeNumerically(">", 0)) // Assert on i to avoid compiler optimization
+
+		time2 := ProcTime{}
+		err = time2.Get(os.Getpid())
+		Expect(err).ToNot(HaveOccurred())
+		Expect(time2.User).To(BeNumerically(">", 0))
+		Expect(time2.Sys).To(BeNumerically(">", 0))
+
+		err = time2.Diff(&time1)
+		if runtime.GOOS == "darwin" {
+			Expect(err).To(Equal(ErrNotImplemented))
+		} else if runtime.GOOS == "windows" {
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
I tried, but it was really flakey, which is why it doesn't assert. I'll add a comment to that effect.